File: nxn/src/lexer/token.rs
#[derive(Debug, Default, Clone, Copy, PartialEq)]
pub enum TokenKind<'code> {
    Document(&'code str),
    Comment(&'code str),

    Fn,
    Let,

    Lparen,
    Rparen,
    Lbrace,
    Rbrace,
    EqualEqual,
    Equal,
    Semicolon,

    Unit,
    Int,

    IntVal(&'code str),
    IdVal(&'code str),

    #[default]
    None,
}

#[derive(Debug, Default, Clone, Copy)]
pub struct Token<'code> {
    pub kind: TokenKind<'code>,
    pub span: [u32; 2],
}

File: src/lexer/token.zig
const IndexSpan = @import("../core/indexSpan.zig").IndexSpan;

pub const Token = struct {
    kind: TokenKind,
    span: IndexSpan,
};

pub const TokenKind = enum(c_int) {
    SPACE = 0,
    DOCUMENT,
    COMMENT,

    // Specifiers
    CON,
    MUT,
    SET,

    // Entities
    FN,
    ENUM,
    UNION,
    STRUCT,

    // Statements
    LET,
    RETURN,

    // Operators
    BREAK,
    CONTINUE,

    // Symbols
    LBRACE,
    RBRACE,
    LBRACK,
    RBRACK,
    LPAREN,
    RPAREN,

    // Operators
    EQUAL,
    EQUALEQUAL,
    COMMA,
    COLON,
    SEMICOLON,

    // Types
    VOID,
    INT,
    FLOAT,

    // Values
    VOIDVAL,
    INTVAL,
    FLOATVAL,
    IDVAL,

    None,
};
